nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -935,71 +943,93 @@ in {
               try_files $uri $uri/ =404;
             '';
           };
-          "~ ^/(?:build|tests|config|lib|3rdparty|templates|data)(?:$|/)".extraConfig = ''
-            return 404;
-          '';
-          "~ ^/(?:\\.(?!well-known)|autotest|occ|issue|indie|db_|console)".extraConfig = ''
-            return 404;
-          '';
-          "~ ^\\/(?:index|remote|public|cron|core\\/ajax\\/update|status|ocs\\/v[12]|updater\\/.+|oc[ms]-provider\\/.+|.+\\/richdocumentscode\\/proxy)\\.php(?:$|\\/)" = {
-            priority = 500;
+          "^~ ${webrootOrSlash}" = {
```

Comment:
I don't really understand why that's necessary: what's wrong with replacing the `/` in each location-definition with `${webrootOrSlash}`?

This just makes the diff unnecessarily hard to review and also makes it harder to understand why certain lines were changed in the future.

Reply:
assuming `overwriteWebroot == null`:
- using ${webrootOrSlash} everywhere we would end up with locations with two starting slashes
- using ${webroot} everywhere we would end up with empty locations

Reply:
the problem comes down to making the configuration work wether the `overwriteWebroot` option is defined or not.
I don't think it's a good solution, but we could keep the current nginx configuration, writing a new one for the `overwriteWebroot` option and splitting between the two cases. But that implies maitaining essentialy the same configuration twice.

Reply:
OK, fair with sticking to what upstream recommends, let's do it like that, then.